dragonfly_grasshopper/src/DF Separate Model.py

```diff
@@ -0,0 +1,37 @@
+# Dragonfly: A Plugin for Environmental Analysis (GPL)
+# This file is part of Dragonfly.
+#
+# Copyright (c) 2019, Ladybug Tools.
+# You should have received a copy of the GNU General Public License
+# along with Dragonfly; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+"""
+Separate a Dragonfly Model object into all of its constituent Dragonfly objects.
+-
+
+    Args:
+        _model: A Dragonfly Model to be separated into into its constituent objects.
+    
+    Returns:
+        buildings: All of the Building objects contained within the input Model.
+        context: All of the ContextShade objects within the input Model.
+"""
+
+ghenv.Component.Name = "DF Separate Model"
+ghenv.Component.NickName = 'SeparateModel'
```

Comment:
I think we should be consistent and use DeconstructModel here unless we are going to use Separate* in other components too.

Reply:
So there is a deliberate distinction that I have been making between "deconstruction" and "separation" in the naming of the components.  Deconstruction describes the breaking of parent-child object relationships in a way that takes some effort to reassemble the original object (like the way the [HB Deconstruct Object](https://github.com/ladybug-tools/honeybee-grasshopper-core/blob/master/honeybee_grasshopper_core/src/HB%20Deconstruct%20Object.py)) component works.  Separation on the other hand just involves separating out the coarsest objects within a Model and not breaking any parent-child object relationships (as you see here in [HB Separate Model component](https://github.com/ladybug-tools/honeybee-grasshopper-core/blob/master/honeybee_grasshopper_core/src/HB%20Separate%20Model.py)).

So, by this definition, the deconstruction of a Honeybee Model would involve things like the gathering of all Face objects across the model into one list (regardless of whether they are a part of a Room or they are orphaned, thereby breaking a parent-child object relationship between certain Faces and Rooms). On the other hand, the separation of a Model preserves all parent-child object relationships and just separates out the orphaned objects.

I realize the distinction is subtle and let me know if you feel that it's too subtle to be useful.  If so, I can change the name of the component here.

Reply:
I see the logic now and it is consistent which was my main concern. But maybe I should ask in what way do you think this separation will improve the user experience? Do you think that this brings more clarity to what the component does?

Reply:
The main reason why I made this distinction is because it allows us to potentially offer a component down the line that actually deconstructs the model (not just separates it). If you think that true Model deconstruction isn't something that we will ever want to offer in a Grasshopper component, then we can just call this one "Deconstruct Model". 

I don't really see myself needing a component to do true deconstruction of a Model but, then again, that's possibly because I'm thinking about energy model applications and, for those, I usually want to keep the properties of the Room and not deconstruct it into un-simulate-able Faces. Do you see any cases where you need all of the Faces in a Model for Radiance applications?

Reply:
I don't think we will need a new component. With the current set-up one can use the deconstruct object to deconstruct the outputs of the separate components. Right? I'm fine to leave it as is. It will most likely be fine either way.